src/core/diffusion_grid.h

```diff
@@ -585,7 +584,7 @@ class DiffusionGrid {
     const double ibl2 = 1 / (box_length_ * box_length_);
     const double d = 1 - dc_[0];
     double step = diffusion_step_;
-    double h = (((double)dt_)/((double)step));
+    double h = (1.0*dt_)/(1.0*step);
```

Comment:
Why are you multiplying by 1.0? Both dt_ and step are already doubles. Same goes for all other occurances in this file.

Reply:
I had assumed this was part of the coding style to enforce floating point operations as it is also done here : 
    https://github.com/BioDynaMo/biodynamo/blob/1f454e0cb10c168c592850803bffe3fb1b4f76a2/src/core/diffusion_grid.h#L103

Reply:
No it's not necessary. I think this remained from the time that box_length_ was an integer value, so even here we don't need the 1.0. Could you please also remove the multiplications by 1.0 from here? You only would need to cast specifically to a double when you do for example a division by two integers (and you expect a decimal answer).